Kudu.Core/SiteExtensions/FeedExtensions.cs

```diff
@@ -161,5 +162,20 @@ public static class FeedExtensions
                 }
             }
         }
+
+        private static async Task<T> GetResourceAndValidateAsync<T>(this SourceRepository srcRepo) where T : class, INuGetResource
+        {
```

Comment:
if we do TryGetResource, when caller see a null provider return. might either return empty result or throw an exception.

return empty result might not be a good idea, since there is something wrong with data, we will need to let the user know. so either way would prefer to throw an exception, and let SiteExtensionController or SIteExtensionManager to catch the exception and return proper error message to user.